### PR TITLE
Add missing config options #121

### LINKED ISSUE #121: A number of config options are missing
When attempting to manage chrony I noticed there are many valid config options that are missing. Namely,

  * allow
  * logchange
  * minsamples
  * minsources

My suspicion is that there are others. I'd like to make a PR to address at least these items though I think it's worth looking into adding a parameter such as `chrony::config_extras` or something similar where a user could pass key/value pairs directly into the resulting `chrony.yaml`

PR to follow adding class params for the 4 mentioned above.

### PR DESCRIPTION
**Pull request by kendall-moore**
#### Pull Request (PR) description

The following options have now been added:
* allow
* deny
* logchange
* minsamples
* minsources

#### This Pull Request (PR) fixes the following issues

Fixes #121

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -230,6 +246,9 @@
   Chrony::Servers $pools                                           = {},
   Numeric $makestep_seconds                                        = 10,
   Integer $makestep_updates                                        = 3,
+  Array[String] $allow                                             = [],
```
Comment (bastelfreak) [inline manifests/init.pp:249]: I think we don't need to allow empty strings in the arrays?
```suggestion
  Array[String[1]] $allow                                          = [],
```

File: manifests/init.pp [hunk 2]
```diff
@@ -230,6 +246,9 @@
...
+  Array[String] $deny                                              = [],
```
Comment (bastelfreak) [inline manifests/init.pp:250]:
```suggestion
  Array[String[1]] $deny                                           = [],
```

File: manifests/init.pp [hunk 3]
```diff
@@ -230,6 +246,9 @@
...
+  Optional[Integer] $minsources                                    = undef,
```
Comment (bastelfreak) [inline manifests/init.pp:251]: can we restrict the values here a bit? Are negative values or 0 allowed?

File: manifests/init.pp [hunk 4]
```diff
@@ -247,6 +266,7 @@
   Optional[Float] $maxdistance                                     = undef,
   Optional[Float] $maxslewrate                                     = undef,
   Optional[Float] $maxupdateskew                                   = undef,
+  Optional[Integer] $minsamples                                    = undef,
```
Comment (bastelfreak) [inline manifests/init.pp:269]: same here, can we restrict the values a bit?

**Comment by bastelfreak** [comment]
Hi @kendall-moore , thanks for the PR. Can you add some spec tests for it?

**Comment by kendall-moore** [comment]
Hi @bastelfreak , thanks for the review! I can definitely add some spec tests, I'll attempt to do that this evening. I'll also check the rest of your suggestions here as soon as I can.

Updates to follow, thanks again!

**Comment by bastelfreak** [comment]
@kendall-moore ping :)

**Comment by jcpunk** [comment]
I believe a version of this is now merged and this request is obsolete.

**Pull request closed without merging**